src/stream/src/executor/dynamic_filter.rs

```diff
@@ -29,27 +30,144 @@ use risingwave_expr::expr::expr_binary_nonnull::new_binary_expr;
 use risingwave_expr::expr::{BoxedExpression, InputRefExpression, LiteralExpression};
 use risingwave_pb::expr::expr_node::Type as ExprNodeType;
 use risingwave_pb::expr::expr_node::Type::*;
+use risingwave_storage::table::state_table::StateTable;
+use risingwave_storage::StateStore;
 
 use super::barrier_align::*;
-use super::error::StreamExecutorError;
+use super::error::{StreamExecutorError, StreamExecutorResult};
 use super::monitor::StreamingMetrics;
 use super::{BoxedExecutor, BoxedMessageStream, Executor, Message, PkIndices, PkIndicesRef};
 use crate::common::StreamChunkBuilder;
 use crate::executor::PROCESSING_WINDOW_SIZE;
 
-pub struct DynamicFilterExecutor {
+pub struct RangeCache<S: StateStore> {
```

Comment:
Now it is not `pub`?

Reply:
I guess I should move it to managed_state. There's not much in common to group it all there but I guess it's the convention so far.